EPCPortalWeb/Models/DataHandlerModel.cs

```diff
@@ -10,7 +10,20 @@ public class DataHandlerModel
         public string Postcode { get; set; }
         public int HouseNumber { get; set; }
 
+        public string APIAddress { get; set; }
 
+        public List<string> FilterAddressesByHouseNumber(int houseNumber, List<string> listOfAPIAddresses)
+        {
+            List<string> newListOfFilteredAddresses = new List<string>();
+            foreach (var address in listOfAPIAddresses)
+            {
+                if (address.Contains(houseNumber.ToString()))
+                {
+                    newListOfFilteredAddresses.Add(address);
+                }
+            }
+            return newListOfFilteredAddresses;
```

Comment:
This is written really clearly, easy to follow.

I believe it could be made shorter using the .Where LINQ method, e.g., return listOfAPIAddresses.Where(address contains house number).